wpiutil/src/main/native/linux/AvahiClient.cpp

```diff
@@ -0,0 +1,82 @@
+#include "AvahiClient.h"
+#include "dlfcn.h"
+#include <wpi/mutex.h>
+#include <thread>
+
+using namespace wpi;
+
+#define AvahiFunctionLoad(snake_name) \
+  snake_name =                        \
+      reinterpret_cast<snake_name##_func>(dlsym(lib, "avahi_" #snake_name))
+
+AvahiFunctionTable::AvahiFunctionTable() {
+  void* lib = dlopen("libavahi-common.so.3", RTLD_LAZY);

```

Comment:
This should probably just exit early if dlopen returns NULL.

Reply:
Done